dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java

```diff
@@ -346,41 +347,46 @@ private void processMimeMessage (MimeMessage mime) throws Exception {
-            }
+            try {
+                JournalUtils.writeManuscriptToDB(manuscript);
+                LOGGER.debug("this ms has status " + manuscript.getStatus());
+                Boolean approved = null;
+
+                if (manuscript.isAccepted()) {
+                    approved = true;
+                } else if (manuscript.isRejected()) {
+                    approved = false;
+                } else if (manuscript.isNeedsRevision()) {
+                    approved = false;
+                } else if (manuscript.isPublished()) {
+                    approved = true;
+                }
 
-            // if the status was "submitted," approved will still be null and we won't try to process any items.
-            if (approved != null) {
-                DSpaceKernelImpl kernelImpl = null;
-                try {
-                    kernelImpl = DSpaceKernelInit.getKernel(null);
-                    if (!kernelImpl.isRunning()) {
-                        kernelImpl.start(ConfigurationManager.getProperty("dspace.dir"));
-                    }
-                } catch (Exception ex) {
-                    // Failed to start so destroy it and log and throw an exception
+                // if the status was "submitted," approved will still be null and we won't try to process any items.
+                if (approved != null) {
+                    DSpaceKernelImpl kernelImpl = null;
                     try {
-                        if(kernelImpl != null) {
-                            kernelImpl.destroy();
+                        kernelImpl = DSpaceKernelInit.getKernel(null);
+                        if (!kernelImpl.isRunning()) {
+                            kernelImpl.start(ConfigurationManager.getProperty("dspace.dir"));
                         }
-                    } catch (Exception e1) {
-                        // Nothing to do
+                    } catch (Exception ex) {
+                        // Failed to start so destroy it and log and throw an exception
+                        try {
+                            if (kernelImpl != null) {
+                                kernelImpl.destroy();
+                            }
+                        } catch (Exception e1) {
+                            // Nothing to do
+                        }
+                        LOGGER.error("Error Initializing DSpace kernel in ManuscriptReviewStatusChangeHandler", ex);
                     }
-                    LOGGER.error("Error Initializing DSpace kernel in ManuscriptReviewStatusChangeHandler", ex);
-                }
 
-                ApproveRejectReviewItem.reviewManuscript(manuscript);
+                    ApproveRejectReviewItem.reviewManuscript(manuscript);
+                }
+            } catch (StorageException e) {
+                LOGGER.error("failed to write ms " + manuscript.getManuscriptId());
+                throw new SubmissionException("failed to write ms " + manuscript.getManuscriptId());
```

Comment:
Add another parameter with the StorageException, like this:
.....anuscriptId(), e);

That way, when it is thrown, you will see both exceptions, linked with a statement that the SubmissionException was caused by the StorageException.